src/frontend/src/binder/expr/function.rs

```diff
@@ -591,14 +591,11 @@ impl Binder {
                 (
                     "array_ndims",
                     guard_by_len(1, raw(|_binder, inputs| {
-                        let input = &inputs[0];
-                        if input.is_untyped() {
-                            return Err(ErrorCode::BindError("could not determine polymorphic type because input has type unknown".into()).into());
-                        }
-                        match input.return_type().array_ndims() {
-                            0 => Err(ErrorCode::BindError("array_ndims expects an array".into()).into()),
-                            n => Ok(ExprImpl::literal_int(n.try_into().map_err(|_| ErrorCode::BindError("array_ndims integer overflow".into()))?))
-                        }
+                        inputs[0].ensure_array_type()?;
+
+                        let n = inputs[0].return_type().array_ndims()
```

Comment:
nit: `array_ndims` won't return 0 now? We can panic inside it.

Reply:
For now, yes, because this is the only caller. But I am considering to consolidate it with `unnest_list` to a single function that converts `List(List(List(Date)))` to `(Date, 3)`, which may be useful for #10168 or similar cases. And then we can see whether panicking is better.